### PR TITLE
pythonPackages.llfuse: darwin support, use in borgbackup

### PR DESCRIPTION
**Pull request by thefloweringash**
###### Motivation for this change

Enabling `borg mount` on macOS

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/llfuse/default.nix [hunk 1]
```diff
@@ -11,9 +11,22 @@ buildPythonPackage rec {
     sha256 = "1j9fzxpgmb4rxxyl9jcf84zvznhgi3hnh4hg5vb0qaslxkvng8ii";
   };
 
+  patches = [
+    # Assumes that the osxfuse kernel extension is installed to test
+    # against. Due to an implementation bug in the tests, actually
+    # errors white loading the test suite.
```
Comment (bjornfor) [inline pkgs/development/python-modules/llfuse/default.nix]: errors whilte -> errors while?

Comment (thefloweringash) [reply pkgs/development/python-modules/llfuse/default.nix]: Yeah, oops.

Comment (thefloweringash) [reply pkgs/development/python-modules/llfuse/default.nix]: Yeah, oops.

**Comment by thefloweringash** [comment]
@GrahamcOfBorg build borgbackup python2Packages.llfuse python3Packages.llfuse

File: pkgs/development/python-modules/llfuse/default.nix [hunk 2]
```diff
@@ -11,9 +11,22 @@ buildPythonPackage rec {
...
+  patches = [
+    # Assumes that the osxfuse kernel extension is installed to test
+    # against. Due to an implementation bug in the tests, actually
+    # errors while loading the test suite.
+    ./assume-no-osxfuse.patch
+
+    # Tests features that require /proc
+    ./skip-test_sup_groups-darwin.patch
+  ];
+
```
Comment (flokli) [inline pkgs/development/python-modules/llfuse/default.nix]: Are these bugs upstream? We usally try to only ship NixOS-specific patches in nixpkgs, and `fetchpatch` patches from PRs instead…

Comment (thefloweringash) [reply pkgs/development/python-modules/llfuse/default.nix]: Upstreamed the patches at https://github.com/python-llfuse/python-llfuse/pull/23, and removed the need for nixpkgs specific packages entirely by using pytest as it should be used.

**Review by LnL7 — APPROVED** [review]

**Pull request merged**